bootloader/test_infra/src/main.rs

```diff
@@ -0,0 +1,166 @@
+        gas_limit: 80_000_000,
+        execution_mode: TxExecutionMode::VerifyExecute,
+        default_validation_computational_gas_limit: 80_000_000,
+        chain_id: zksync_types::L2ChainId(299),
+    };
+
+    let mut l1_batch_env = L1BatchEnv {
+        previous_batch_hash: None,
+        number: L1BatchNumber::from(13),
+        timestamp: 14,
+        l1_gas_price: 250_000_000,
+        fair_l2_gas_price: 250_000_000,
+        fee_account: Address::default(),
+
+        enforced_base_fee: None,
+        first_l2_block: L2BlockEnv {
+            number: 1,
+            timestamp: 15,
+            prev_block_hash: legacy_miniblock_hash(MiniblockNumber(0)),
+            max_virtual_blocks_to_create: 0,
+        },
+    };
+
+    // First - get the number of tests.
+    let test_count = {
+        // FIXME: this is still taking System Contracts from ZKSYNC_HOME
+        let storage: StoragePtr<StorageView<InMemoryStorage>> =
+            StorageView::new(InMemoryStorage::with_system_contracts(hash_bytecode)).to_rc_ptr();
+
+        let mut vm = Vm::new(
+            l1_batch_env.clone(),
+            system_env.clone(),
+            storage.clone(),
+            HistoryDisabled,
+        );
+
+        let test_count = Arc::new(OnceCell::default());
+        let custom_tracers = vec![Box::new(TestCountTracer::new(test_count.clone()))
+            as Box<dyn VmTracer<StorageView<InMemoryStorage>, HistoryDisabled>>];
+
+        // We're using a TestCountTracer (and passing 0 as fee account) - this should cause the bootloader
+        // test framework to report number of tests via VM hook.
+        vm.inspect(custom_tracers, VmExecutionMode::Bootloader);
+
+        test_count.get().unwrap().clone()
+    };
+    println!(" ==== Running {} tests ====", test_count);
+
+    let mut tests_failed: u32 = 0;
+
+    // Now we iterate over the tests.
+    for test_id in 1..=test_count {
+        println!("\n === Running test {}", test_id);
+
+        let storage: StoragePtr<StorageView<InMemoryStorage>> =
+            StorageView::new(InMemoryStorage::with_system_contracts(hash_bytecode)).to_rc_ptr();
+
+        // We are passing id of the test in location (0) where we normally put the operator.
+        // This is then picked up by the testing framework.
+        l1_batch_env.fee_account = zksync_types::H160::from(u256_to_h256(U256::from(test_id)));
+        let mut vm = Vm::new(
+            l1_batch_env.clone(),
+            system_env.clone(),
+            storage.clone(),
+            HistoryDisabled,
+        );
+        let test_result = Arc::new(OnceCell::default());
+
+        let custom_tracers = vec![Box::new(BootloaderTestTracer::new(test_result.clone()))
+            as Box<dyn VmTracer<StorageView<InMemoryStorage>, HistoryDisabled>>];
+
+        // Let's insert transactions into slots. They are not executed, but the tests can run functions against them.
+        let json_str = include_str!("test_transactions/0.json");
```

Comment:
my plan was, that we add these into the test_transactions as json files - and then add them all here.

They will come as different indexes, and then tests can access them when they want.
In theory this would work to up to 1k transactions (which should be enough for now)